pkg/controllers/vdb/install_reconciler.go

```diff
@@ -101,6 +102,17 @@ func (d *InstallReconciler) installForAdmintools(ctx context.Context) (ctrl.Resu
 		// hosts to the config, we have to know that a re_ip will succeed.
 		d.addHostsToATConf,
 	}
+	// generateHTTPCerts setup is only required for versions before 24.1.0 of the database.
+	vinf, err := d.Vdb.MakeVersionInfoCheck()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if vinf.IsOlder(vapi.VcluseropsAsDefaultDeploymentMethodMinVersion) {
```

Comment:
@Matt I think the operator should either always generate the certs or do it if image version >= "v23.4.0" for admintools. Because before this change, the operator never generated certs for any admintools deployment. So for an upgrade case like 24.1.0(admintools)-->24.2.0(vclusterops) we are still going to need certs.